zhaquirks/nimly/lock.py

```diff
@@ -25,11 +32,121 @@
 )
 
 
+class NimlyDoorLock(CustomCluster, DoorLock):
+    """Nimly Door Lock cluster."""
+
+    class AttributeDefs(DoorLock.AttributeDefs):
+        """Nimly Door Lock attribute definitions."""
+
+        nimly_last_lock_unlock_source: Final = ZCLAttributeDef(
+            id=0x100,
+            type=t.bitmap32,
+            access="r",
+            is_manufacturer_specific=True,
+        )
+        nimly_last_pin_code: Final = ZCLAttributeDef(
+            id=0x101,
+            type=t.LVBytes,
+            access="r",
+            is_manufacturer_specific=True,
+        )
+
+
+def last_action_source_converter(value: int) -> str:
+    """Extract last action source value."""
+    value_hex = hex(value)[2:].zfill(8)
+    sources = {
+        "00": "zigbee",
+        "02": "keypad",
+        "03": "fingerprint",
+        "04": "rfid",
+        "0a": "self",
+    }
+    return sources.get(value_hex[0:2])
+
+
+def last_action_converter(value: int) -> str:
+    """Extract last action value."""
+    value_hex = hex(value)[2:].zfill(8)
+    actions = {
+        "01": "lock",
+        "02": "unlock",
+    }
+    return actions.get(value_hex[2:4])
+
+
+def last_action_user_converter(value: int) -> int:
+    """Extract last action user value."""
+    value_hex = hex(value)[2:].zfill(8)
+    user_id = int(value_hex[4:8], 16)
+    return user_id
+
+
+def last_pin_code_converter(value: bytes) -> str:
+    """Extract last used pin code."""
+    return "".join(hex(b)[2:] for b in value)
+
+
 (
     QuirkBuilder(NIMLY, "EasyFingerTouch")
     .applies_to(NIMLY, "EasyCodeTouch")
     .applies_to(NIMLY, "easyCodeTouch_v1")
     .node_descriptor(NIMLY_LOCK_NODE_DESCRIPTOR)
+    .sensor(
+        endpoint_id=11,
+        cluster_id=NimlyDoorLock.cluster_id,
+        attribute_name=NimlyDoorLock.AttributeDefs.nimly_last_lock_unlock_source.name,
+        translation_key="last_action_source",
+        fallback_name="Last action source",
```

Comment:
Can we move `translation_key` and then `fallback_name` to the bottom (so as the last argument per method)?
That matches what we do for all other v2 quirks.